Libraries/Azure.psm1

```diff
@@ -2363,18 +2363,21 @@ function Check-AzureVmKernelPanic {
         Downloads the Azure Boot diagnostics and checks if they contain kernel panic or RIPs.
     #>
     param(
-        $Vm
+        $Vm,
+        $KernelLogs
     )
 
-    $bootDiagnosticFile = "$LogDir\$($vm.RoleName)-SSH-Fail-Boot-Logs.txt"
-    $diagStatus = Get-AzureBootDiagnostics -Vm $vm -BootDiagnosticFile $bootDiagnosticFile
-    if ($diagStatus -and (Test-Path $bootDiagnosticFile)) {
-        $diagFileContent = Get-Content $bootDiagnosticFile
-        if ($diagFileContent -like "*Kernel panic - not syncing:*" -or $diagFileContent -like "*RIP:*" `
-              -or $diagFileContent -like '*grub>*') {
-            return $true
+    if (!$KernelLogs) {
+        $bootDiagnosticFile = "$LogDir\$($vm.RoleName)-SSH-Fail-Boot-Logs.txt"
+        $diagStatus = Get-AzureBootDiagnostics -Vm $vm -BootDiagnosticFile $bootDiagnosticFile
+        if ($diagStatus -and (Test-Path $bootDiagnosticFile)) {
+             $KernelLogs = Get-Content $bootDiagnosticFile
         }
     }
+    if ($KernelLogs -like "*Kernel panic - not syncing:*" -or $KernelLogs -like "*RIP:*" `
```

Comment:
Recently we've implemented a way to put text identifiers in one place.
https://github.com/LIS/LISAv2/blob/master/XML/Other/text-identifiers.xml
And fetch it for comparison, from this file.

In your case also, you can do this to detect the call traces.

**Example:** 
```
.
.
.
            <keyword>Successfully reported handler status</keyword> <!-- Azure Linux Agent Version: WALinuxAgent-2.0.16 -->
        </ProvisionComplete>
    </waagent>
    <kernel>
        <Panic>
            <keyword>Kernel panic - not syncing</keyword>
            <keyword>Call Trace</keyword>
            <keyword>rcu_sched self-detected stall on CPU</keyword>
            <keyword>rcu_sched detected stalls on</keyword>
            <keyword>BUG: soft lockup</keyword>
        </Panic>
    </kernel>
```